### PR TITLE
Enable gradients at train start

### LINKED ISSUE #2180: Global Gradient calculation is turned off during validation step.
If an error occurs during the validation step, the tradition calculation is turned off for the runtime, you have to either specifically enable it or restart runtime!

### PR DESCRIPTION
**Pull request by Nilanshrajput**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## What does this PR do?
Fixes #2180 .

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
yup

### CODE REVIEW

**Comment by Nilanshrajput** [comment]
@Borda can you suggest tests, I looked at the tests section but there is use of a model template and to recreate that error we might need to change it/or create a separate one.

**Comment by Borda** [comment]
> @Borda can you suggest tests, I looked at the tests section but there is use of a model template and to recreate that error we might need to change it/or create a separate one.

probably add a custom test callback which will be testing if model gas gradients... 
it would be easier to do after adding these hook in #2217

**Comment by williamFalcon** [comment]
thanks @Nilanshrajput
